app/Jobs/ExportPostsJob.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Ushahidi\App\Jobs;
+
+use Exception;
+use Ushahidi\Core\Usecase\Export\Job\PostCount;
+use Ushahidi\Core\Entity\ExportJob;
+use Ushahidi\Core\Entity\ExportJobRepository;
+use Illuminate\Support\Facades\Log;
+
+class ExportPostsJob extends Job
+{
+    protected $batchSize = 200;
+
+    protected $jobId;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($jobId)
+    {
+        $this->jobId = $jobId;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(PostCount $usecase, ExportJobRepository $exportJobRepo)
+    {
+        // Load job
+        $job = $exportJobRepo->get($this->jobId);
+
+        // Get total posts count
+        // @todo this probably doesn't need its own usecase
+        $usecase->setIdentifiers(['id' => $this->jobId]);
+        $results = $usecase->interact();
+        $totalRows = $results[0]['total'];
+
+        Log::debug('Got job count', ['jobId' => $this->jobId, 'results' => $results]);
+
+        // If count is zero. Mark as failed
+        if ($totalRows == 0) {
+            // Set status = failed
+            $job->setState([
+                'total_batches' => 0,
+                'total_rows' => 0,
+                'status' => ExportJob::STATUS_FAILED,
+            ]);
+            $exportJobRepo->update($job);
+            // All done
+            return;
+        }
+
+        $totalBatches = ceil($totalRows / $this->batchSize);
+
+        // Generate other meta data here too? ie. header rows
+        //
+        Log::debug('Queuing batches', ['totalBatches' => $totalBatches]);
+        for ($batchNumber = 0; $batchNumber < $totalBatches; $batchNumber++) {
+            Log::debug('Queuing batch', [
+                'jobId' => $this->jobId,
+                'batchNumber' => $batchNumber,
+                'offset' => $batchNumber * $this->batchSize,
+                'limit' => $this->batchSize
+            ]);
+            dispatch(new ExportPostsBatchJob(
+                $this->jobId,
+                $batchNumber,
+                $batchNumber * $this->batchSize,
+                $this->batchSize,
+                ($batchNumber === 0)
+            ));
+        }
+
+        // Set status = queued
+        $job->setState([
```

Comment:
This has a slightly bizarre result if QUEUE_DRIVER=sync.
The jobs above run before this update. So the final job status = queued, instead of completed.

I can swap the order, but its just a bit weird. @rowasc @kinstelli thoughts?